### DISCUSSION TITLE
Micro SD card not readable from macbook, but works with MMP?

### DISCUSSION THREAD

**Discussion by Pakman450** [OnionUI/Onion#1249] [Support] [answered]
# What happened
Hello, I have this problem in my M1 Macbook pro. My MMP can read and write to the micro SD card. When I insert my card into my macbook, I can `cd` and navigate through my files in the micro sd card (Figure 1).  But when I try to view it on my Finder, it doesn't want to show anything even though it recognizes the sd card (Figure 2). Also, when I tried to copy my files into my main hard drive weird error messages come up (Figure 3)

**Figure 1**
![Screenshot 2023-10-27 at 9 20 03 AM](https://github.com/OnionUI/Onion/assets/69532598/2a6e8a7e-bcf8-4644-9b6a-15e30008b620)

**Figure 2**
![Screenshot 2023-10-27 at 9 20 19 AM](https://github.com/OnionUI/Onion/assets/69532598/dcb58d6a-500f-43e3-a25f-7c54a8b5718e)

**Figure 3**
![Screenshot 2023-10-27 at 9 21 35 AM](https://github.com/OnionUI/Onion/assets/69532598/393ead58-81cc-4101-95b6-8dd40e72c0be)

# How did this happen
When I updated to 4.2.1 over wifi, these weird things happened. I "fixed" it by deleting files I have never seen before, specifically `FSCK[XXXX].REC` files (Can't really be detailed on how I did it. I forgot). Then, when I updated 4.2.2 over wifi, it _**happened again**_. I theorize the updating process over wifi is messing with reading capabilities of the micro SD. 

 I want to reiterate. My experience with MMP and my SD card is great! Nothing is wrong with that. 

If anyone can figure what is going on, Thank you.

**Comment by Phanturian** [comment]
It looks like you are getting permission issues with the MIYOOMINIPL card. Try right clicking (or two finger press on trackpad) the desktop icon for MIYOOMINIPL and Get Info. Check the permissions at the bottom to make sure there are no conflicts. Can you drag/drop or copy anything to the MIYOOMINIPL card? If so, try downloading the latest Onion and manually copy everything over, especially the hidden .tmp_update folder. It's possible you may have deleted hidden files that your Mac needs to recognize the contents of your SD card. If you have a way of getting your roms/saves/configs off the MIYOOMINIPL card, I might do that and then format it, and copy everything back to it.

I'm a Mac user who is constantly updating my Miyoo Mini and Plus systems to the latest Onion version. I personally choose to do it manually because I find it faster, but I have used the OTA updater with no issues. I never use Terminal for any Miyoo SD card operations other than running dot_clean before every eject. I just copy/paste and delete from the SD card using Finder. I hope you get it figured out.

**Comment by dolphin-hacker** [comment]
Same thing happened when I used the OTA update to get 4.2.3 today. My SD card got renamed to UNTITLED, and I couldn't view the file system in Finder, but for all intents and purposes my MacBook Pro M1 is recognizing the card and there aren't any errors or anything.  So I renamed the card to MMP, unmounted it, remounted it, then ran a "First Aid" on the partition: 

<img width="1339" alt="Screenshot 2023-11-06 at 17 06 00" src="https://github.com/OnionUI/Onion/assets/109117078/4be6e2fe-3645-436d-8a77-bc8bcd936a8b">

Bam all the files are now viewable, and what the heck are these FSCKxxxx files, so weird.  

<img width="1478" alt="Screenshot 2023-11-06 at 17 02 16" src="https://github.com/OnionUI/Onion/assets/109117078/75e14dc4-ef42-4880-b639-2d999468065c">

**Comment by chrischard** [comment, marked as answer]
Exact same issue for me. I updated Onion on my Miyoo Mini Plus via OTA yesterday to 4.2.1 and now my MacBook Pro M1 can't see any files on the SD Card. Running a First Aid through Disk Utility fixed it, here's the log from Disk Utility:

```
Running First Aid on “UNTITLED” (disk4s1)

Checking file system and repairing if necessary and if possible.
Volume was successfully unmounted.
Performing fsck_msdos -y /dev/rdisk4s1
** /dev/rdisk4s1

** Phase 1 - Preparing FAT

** Phase 2 - Checking Directories

Warning: /FSCK0000.REC starts with free cluster

Truncate? yes

Warning: /FSCK0001.REC starts with free cluster

Truncate? yes

** Phase 3 - Checking for Orphan Clusters

Warning: 12281 files, 47868288 KiB free (1495884 clusters)

Warning: 

***** FILE SYSTEM WAS MODIFIED *****

File system check exit code is 0.
Restoring the original state found as mounted.

Operation successful.

```

**Comment by emperordahc** [comment]
Same exact issue on my Macbook Pro and my MM+. SD card runs just fine in my MM+, but when I pop it in my Mac, it's blank. I run Disk Utility multiple times, and after eject/reinserting multiple times, files suddenly show up. Same huge list of .REC files in the root, too.

As mentioned by another user, for general use I just drag/drop files in Finder and then use Terminal dot_clean before ejecting.

These issues started happening after using the OTA updates. I don't know if it's related, but I noticed the updater seems to want to "fix" something on the SD card every time it updates.
